### PR TITLE
Residual attention network

### PR DESCRIPTION
**Pull request by PistonY**
Add ResidualAttentionNetwork.
params still training.
It''s based on this [projects](https://github.com/PistonY/ResidualAttentionNetwork)

### CODE REVIEW

**Comment by zhreshold** [comment]
Fantastic, is CIFAR the only target provided?

**Comment by PistonY** [comment]
I still training imagenet and I will add AtteneionResNext, AttentionInception latter.
I just have a try to pr.

**Comment by PistonY** [comment]
This net is not one network adapt all input shapes,the inputs shape is almost fixed cause the number of downsampling&upsampling details is different.
AttentionXX_32input -- input size of 32x32
AttentionXX               -- ipnuts size of 224 or 320
AttentionXX_448input -- input size of 448 or 512(I'm not sure it work or not,may be delete in the future.)

File: gluoncv/model_zoo/residual_attentionnet.py [hunk 1]
```diff
@@ -0,0 +1,860 @@
...
+        out_softmax1 = self.softmax1_blocks(x)
+        out_softmax2 = self.softmax2_blocks(out_softmax1)
+        out = F.elemwise_add(F.ones_like(out_softmax2), out_softmax2)
+        out = F.elemwise_mul(out, out_trunk)
+        out_last = self.last_blocks(out)
+
+        return out_last
+
+
+class ResidualAttentionModel_448input(nn.HybridBlock):
```
Comment (hetong007) [inline gluoncv/model_zoo/residual_attentionnet.py]: Please try the best to parameterize the difference among the networks, which means that we only need one definition of the network, and the difference (for example the number of channels) can be input arguments.

One example: https://github.com/hetong007/gluon-cv/blob/master/gluoncv/model_zoo/cifarresnet.py#L245

Comment (PistonY) [reply gluoncv/model_zoo/residual_attentionnet.py]: OK,I'll refactor code.

**Comment by mli** [comment]
Job PR-382-5 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-382/5/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-382-7 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-382/7/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-382-8 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-382/8/index.html 
Code coverage of this PR:  vs. Master:

**Comment by PistonY** [comment]
Almost finish...
AttentionResNeXt will be in next pr.
AttentionInception may be in next pr.
Using p, t, r control net scale will be in next pr.(paper mentioned)
How to update param is [discussing](https://discuss.gluon.ai/t/topic/8087/34).
Any issues please write up here or Gluon forum.

**Comment by mli** [comment]
Job PR-382-10 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-382/10/index.html 
Code coverage of this PR:  vs. Master:

**Comment by PistonY** [comment]
This is not my bugs....

**Comment by PistonY** [comment]
This is not my bugs....

**Comment by zhreshold** [comment]
try rebase with upstream master

**Comment by PistonY** [comment]
好像操作的不太对=-=

**Comment by PistonY** [comment]
这回好像对了==

**Comment by PistonY** [comment]
要不我删了重新pr一遍吧......

**Comment by mli** [comment]
Job PR-382-12 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-382/12/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-382-14 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-382/14/index.html 
Code coverage of this PR:  vs. Master:

**Comment by zhreshold** [comment]
@PistonY Well, thanks for the efforts, but now it has many unrelated commits inside. 
Can you follow the instructions to make the PR cleaner? It is almost impossible for us to review the changes.

You can reopen another PR if it already got messed up.

**Comment by PistonY** [comment]
OK that.I'll reopen another one.

**Pull request closed without merging**